Druid/DRCogitCompilationUnit.class.st

```diff
@@ -93,76 +92,95 @@ DRPrimitiveCompilationUnit >> compilePrimitive: aDRPrimitiveObject [
 ]
 
 { #category : #private }
-DRPrimitiveCompilationUnit >> dispatchTableBuilder [
+DRCogitCompilationUnit >> dispatchTableBuilder [
 	" Answer a <PCGMethodNode> representing the source code of the #initializePrimitiveTable compatible with Cogit "
 
 	^ dispatchTableBuilder
-		ifNil: [ dispatchTableBuilder := self dispatchTableBuilderClass primitiveTableMethodFrom: self ]
+		ifNil: [ dispatchTableBuilder := self dispatchTableBuilderClass fromCompilationUnit: self ]
 ]
 
 { #category : #private }
-DRPrimitiveCompilationUnit >> dispatchTableBuilder: aDRDispatchTableBuilder [
+DRCogitCompilationUnit >> dispatchTableBuilder: aDRDispatchTableBuilder [
 	" Set aDRDispatchTableBuilder to the receiver's dispatch table builder "
 
 	dispatchTableBuilder := aDRDispatchTableBuilder
 ]
 
 { #category : #private }
-DRPrimitiveCompilationUnit >> dispatchTableBuilderClass [ 
+DRCogitCompilationUnit >> dispatchTableBuilderClass [ 
 
 	^ DRCogitDispatchTableBuilder
 ]
 
 { #category : #testing }
-DRPrimitiveCompilationUnit >> hasPrimitives [
+DRCogitCompilationUnit >> hasPrimitives [
 	"Answer <true> if the receiver already contains primitives"
 
 	^ self primitives notEmpty
 ]
 
+{ #category : #accessing }
+DRCogitCompilationUnit >> initBytecodeTable [
+
+	^ initBytecodeTable
```

Comment:
Why lazy initialized?
Also, the name of this variable is strange. Why "init"? It either contains a bytecode table, or it does not... It does not contain an init :)